core/drivers/ls_sfp.c

```diff
@@ -0,0 +1,391 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022 Microsoft
+ *
+ * Driver for the NXP LX2160A-series Security Fuse Processor (SFP).
+ */
+
+#include <assert.h>
+#include <drivers/ls_sfp.h>
+#include <io.h>
+#include <kernel/boot.h>
+#include <kernel/dt.h>
+#include <kernel/tee_time.h>
+#include <libfdt.h>
+#include <mm/core_memprot.h>
+#include <util.h>
+
+/**
+ * struct ls_sfp_registers - Memory map of the SFP registers.
+ * rsvd0[0x8]:		Reserved.
+ * @ingr:		Instruction Register.
+ * @svhesr:		Secret Value Hamming Error Status Registers.
+ * @sfpcr:		SFP Configuration Register.
+ * @rsvd1[0x3]:		Reserved.
+ * @version:		SFP Version Register.
+ * @rsvd2[0x71]:	Reserved.
+ * @ospr0:		OEM Security Policy Register 0.
+ * @ospr1:		OEM Security Policy Register 1.
+ * @dcvr0:		Debug Challenge Value Register 0.
+ * @dcvr1:		Debug Challenge Value Register 1.
+ * @drvr0:		Debug Response Value Register 0.
+ * @drvr1:		Debug Response Value Register 1
+ * @fswpr:		Factory Section Write Protect Register.
+ * @fuidr0:		Factory Unique ID Register 0.
+ * @fuidr1:		Factory Unique ID Register 1.
+ * @isbccr:		ISBC Configuration Register.
+ * @fspfr[0x3]:		Factory Scratch Pad Fuse Registers.
+ * @otpmkr[0x8]:	One Time Programmable Master Key.
+ * @srkhr[0x8]:		Super Root Key Hash.
+ * @ouidr[0x5]:		OEM Unique ID Scratch Pad Fuse Registers.
+ */
+static struct ls_sfp_registers {
+	uint32_t rsvd0[0x8];	/* 0x000 */
+	uint32_t ingr;		/* 0x020 */
+	uint32_t svhesr;	/* 0x024 */
+	uint32_t sfpcr;		/* 0x028 */
+	uint32_t rsvd1[0x3];	/* 0x02C */
+	uint32_t version;	/* 0x038 */
+	uint32_t rsvd2[0x71];	/* 0x03C */
+	uint32_t ospr0;		/* 0x200 */
+	uint32_t ospr1;		/* 0x204 */
+	uint32_t dcvr0;		/* 0x208 */
+	uint32_t dcvr1;		/* 0x20C */
+	uint32_t drvr0;		/* 0x210 */
+	uint32_t drvr1;		/* 0x214 */
+	uint32_t fswpr;		/* 0x218 */
+	uint32_t fuidr0;	/* 0x21C */
+	uint32_t fuidr1;	/* 0x220 */
+	uint32_t isbccr;	/* 0x224 */
+	uint32_t fspfr[0x3];	/* 0x228 */
+	uint32_t otpmkr[0x8];	/* 0x234 */
+	uint32_t srkhr[0x8];	/* 0x254 */
+	uint32_t ouidr[0x5];	/* 0x274 */
+} *sfp_regs;
+
+/**
+ * struct ls_gpio_info - Data struct containing GPIO specific information.
+ * @gpio_pin:		GPIO pin number.
+ * @ls_gpio_chip_data:	GPIO controller instance data.
+ */
+static struct ls_gpio_info {
+	uint32_t gpio_pin;
+	struct ls_gpio_chip_data gpio_chip;
+} gpio_info;
+
+/**
+ * ls_sfp_init() - Get SFP info from the embedded device tree and initialize
+ *		   sfp_regs and gpio_info.
+ *
+ * Return:	TEE_SUCCESS or > 0 on error.
+ */
+static TEE_Result ls_sfp_init(void)
+{
+	size_t size = 0;
+	int32_t node = 0;
```

Comment:
Prefer `int` as this is fdt function's native node type.
Ditto for `rc`, `povdd_node` and `prop_len`.